components/wmtk_components/manifold_extraction/internal/Manifold-extraction.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <cstddef>
+#include <array>
+#include <vector>
+#include <cstdlib>
+#include <random>
+#include <map>
+#include <Eigen/Dense>
+
+
+namespace wmtk {
+using Point3D = std::array<double, 3>;
```

Comment:
Please do not put `using` or `typedef` in header files if not for very good reasons. This using will leak into so much code where it will cause trouble. You can use `using` but please only locally.

Additionally, in the toolkit you should use `Eigen::Vector3d` for a 3d point.